Source/multi.cpp

```diff
@@ -724,60 +714,59 @@ int __fastcall NetInit(int bSinglePlayer, int *pfExitProgram)
 		UiData.profilebitmapcallback = UiProfileDraw;
 		UiData.profilecallback = UiProfileCallback;
 		UiData.profilefields = UiProfileGetString();
-		memset(sgbPlayerTurnBitTbl, 0, 4u);
-		gbGameDestroyed = 0;
-		memset(sgbPlayerLeftGameTbl, 0, 4u);
-		memset(sgdwPlayerLeftReasonTbl, 0, 0x10u);
-		memset(sgbSendDeltaTbl, 0, 4u);
-		memset(plr, 0, 0x15360u);
-		memset(sgwPackPlrOffsetTbl, 0, 8u);
+		memset(sgbPlayerTurnBitTbl, 0, sizeof(sgbPlayerTurnBitTbl));
+		gbGameDestroyed = FALSE;
+		memset(sgbPlayerLeftGameTbl, 0, sizeof(sgbPlayerLeftGameTbl));
+		memset(sgdwPlayerLeftReasonTbl, 0, sizeof(sgdwPlayerLeftReasonTbl));
+		memset(sgbSendDeltaTbl, 0, sizeof(sgbSendDeltaTbl));
+		memset(plr, 0, sizeof(plr));
+		memset(sgwPackPlrOffsetTbl, 0, sizeof(sgwPackPlrOffsetTbl));
 		SNetSetBasePlayer(0);
-		if (v14)
-			v4 = multi_init_single(&ProgramData, &a2, &UiData);
-		else
-			v4 = multi_init_multi(&ProgramData, &a2, &UiData, a4);
-		if (!v4)
-			return 0;
-		sgbNetInited = 1;
-		sgbTimeout = 0;
+		if (bSinglePlayer) {
+			if (!multi_init_single(&ProgramData, &plrdata, &UiData))
+				return FALSE;
+		} else {
+			if (!multi_init_multi(&ProgramData, &plrdata, &UiData, pfExitProgram))
+				return FALSE;
+		}
+		sgbNetInited = TRUE;
+		sgbTimeout = FALSE;
 		delta_init();
 		InitPlrMsg();
 		buffer_init(&sgHiPriBuf);
 		buffer_init(&sgLoPriBuf);
-		dword_678628 = 0;
+		dword_678628 = FALSE;
 		sync_init();
 		nthread_start(sgbPlayerTurnBitTbl[myplr]);
 		dthread_start();
-		MI_Dummy(0); /* v5 */
+		Multi_Dummy();

```

Comment:
Looks like the original code called a nop function that got merged with `MI_Dummy` (0x440694) by the linker, but the original function didn't take any params and MI_Dummy takes one.

Reply:
Interesting, I wondered what that was there for.

Reply:
I wasn't able to locate the original one in the debug build, NetInit appears to have been altered a lot between 1.00 and 1.09 (probably as part of the reordered multiplayer menu), so the extra function might have been added in that time frame.

Reply:
I saw this the other day when reversing `dx_init`. It also invoked a `MI_Dummy(0)` as a NOP call. However, the function it actually invoked was `break_exception` in the debug release. And `break_exception` takes no parameters. However, the compiler merged functions with empty bodies, so it merged `break_exception` with `MI_Dummy`, but since `MI_Dummy` takes one argument, the decompiler got confused.

Reply:
Nice, I have copied `break_exception` to this PR also, at the very least it's more realistic then the solution I had.

Reply:
Had to change it a bit to get it to compile, and as they say it's not suitable for 64bit systems:
https://docs.microsoft.com/en-us/windows/desktop/debug/using-a-vectored-exception-handler